aiomysql/connection.py

```diff
@@ -786,53 +804,196 @@ def _get_server_information(self):
+        else:  # pragma: no cover
+            # not testing against servers without secure auth (>=5.0)
+            data += authresp + b'\0'
+
+        # Append DB
+        if self._db and self.server_capabilities & CLIENT.CONNECT_WITH_DB:
+            if isinstance(self._db, str):
+                db = self._db.encode('latin1')
             else:
-                raise InterfaceError(self._close_reason)
+                db = self._db
+            data += db + b'\0'
 
-    if PY_341:  # pragma: no branch
-        def __del__(self):
-            if self._writer:
-                warnings.warn("Unclosed connection {!r}".format(self),
-                              ResourceWarning)
-                self.close()
-    Warning = Warning
-    Error = Error
-    InterfaceError = InterfaceError
-    DatabaseError = DatabaseError
-    DataError = DataError
-    OperationalError = OperationalError
-    IntegrityError = IntegrityError
-    InternalError = InternalError
-    ProgrammingError = ProgrammingError
-    NotSupportedError = NotSupportedError
+        if self.server_capabilities & CLIENT.PLUGIN_AUTH:
+            name = auth_plugin
+            if isinstance(name, str):
+                name = name.encode('ascii')
+            data += name + b'\0'
+
+        await self.write_packet(data, no_ack=True)
+        auth_packet = await self._read_packet()
+
+        #
+        # # if authentication method isn't accepted the first byte
+        # # will have the octet 254
+        if auth_packet.is_auth_switch_request():
+            # https://dev.mysql.com/doc/internals/en/
+            # connection-phase-packets.html#packet-Protocol::AuthSwitchRequest
+            auth_packet.read_uint8()  # 0xfe packet identifier
+            plugin_name = auth_packet.read_string()
+            if (self.server_capabilities & CLIENT.PLUGIN_AUTH and
+                    plugin_name is not None):
+                auth_packet = self._process_auth(plugin_name, auth_packet)
+            else:
+                # send legacy handshake
+                data = _scramble_323(self._password.encode('latin1'),
+                                     self.salt) + b'\0'
+                self.write_packet(data)
+                auth_packet = await self._read_packet()
+
+    async def write_packet(self, payload, no_ack=False):
+        """Writes an entire "mysql packet" in its entirety to the network
+        addings its length and sequence number.
+        """
+        # Internal note: when you build packet manually and calls
+        # _write_bytes() directly, you should set self._next_seq_id properly.
+        data = pack_int24(len(payload)) + int2byte(self._next_seq_id) + payload
+        await self.write(data)
+
+        if no_ack:
+            self._next_seq_id = (self._next_seq_id + 1) % 256
+
+    async def write(self, data):
+        self._writer.write(data)
+        try:
+            await self._writer.drain()
+        except ConnectionResetError:
```

Comment:
Why we suppress ConnectionResetError ?

Reply:
Errr, I wrote this PR about 5 months ago, I really cannot remember. I half want to blame the odd switching to SSL part way through. Once sec, I'll go run some tests without it.